server/src/main/scala/org/http4s/server/middleware/CSRF.scala

```diff
@@ -244,6 +247,24 @@ final class CSRF[F[_], G[_]] private[middleware] (
 
 object CSRF {
 
+  final case class CSRFCookieSettings(
```

Comment:
Are we fairly certain that nothing will be added to this in the future?  I'm trying to move away from case classes for config for binary compatibility reasons.  My altnernative has been the server and client builders, which are painful to write, but can support additions in the future without breaking code.

Reply:
+1 on builders. Started on a (much less thorough) builder-like addition here https://github.com/http4s/http4s/pull/2235 until i discovered this PR.

Reply:
I don’t think much more is needed wrt cookie settings. Are you thinking about a builder for the middleware itself @rossabaker ?

Reply:
I was thinking a builder for the config, but I suppose it could be a builder for the middleware.  We merged configuration and builder in the server and client builders.  I haven't fully thought through the ergonomics of it for middleware.  :thinking:

* `CSRF(http)`: defaults
* `CSRF.withCookieName("x-csrf-token").withSecure(true)(http)`

So `CSRF` is a class that is `Http` => `Http`.  Object (or value) `CSRF` is  `CSRF` with default values.  And you can configure things until applying the `http`.

This pattern could apply to a few middlewares, and people are antsy for 0.20.  Maybe we roll with it as is, and introduce this refactoring before 1.0.